cli/command/utils_test.go

```diff
@@ -0,0 +1,86 @@
+package command
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWaitForDeployment(t *testing.T) {
+	defer SetTimeMultiplier(0)()
+
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	deployments := []models.Deployment{
+		{
+			DesiredCount: 1,
+			RunningCount: 1,
+		},
+	}
+
+	expected := &models.Service{
+		Deployments:  deployments,
+		DesiredCount: 1,
+		RunningCount: 1,
+	}
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Times(4).
+		Return(expected, nil)
+
+	result, err := WaitForDeployment(base.Client, "svc_id", time.Second)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, expected, result)
+}
+
+func TestWaitForDeployment_error(t *testing.T) {
+	defer SetTimeMultiplier(0)()
+
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	expected := fmt.Errorf("Error reading service")
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Return(nil, expected)
+
+	if _, err := WaitForDeployment(base.Client, "svc_id", time.Second); err == nil {
+		t.Fatal("Error was nil!")
+	}
+}
+
+func TestWaitForDeployment_timeout(t *testing.T) {
+	defer SetTimeMultiplier(0)()
+
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	deployments := []models.Deployment{
+		{
+			DesiredCount: 1,
+			RunningCount: 0,
+		},
+	}
+
+	expected := &models.Service{
+		Deployments: deployments,
+	}
+
+	base.Client.EXPECT().
```

Comment:
shouldn't we expect this not to execute?

Reply:
@zpatrick This got a little lost as I was moving things about; I assume you're talking about `TestWaitForDeployment_timeout()`, which now lives in `layer0/client/helpers_test.go`.

```
func TestWaitForDeployment_timeout(t *testing.T) {
	defer SetTimeMultiplier(0)()

	ctrl := gomock.NewController(t)
	defer ctrl.Finish()
	client := mock_client.NewMockClient(ctrl)

	deployments := []models.Deployment{
		{
			DesiredCount: 1,
			RunningCount: 0,
		},
	}

	expected := &models.Service{
		Deployments: deployments,
	}

	client.EXPECT().
		ReadService("svc_id").
		Return(expected, nil).
		AnyTimes()

	if _, err := WaitForDeployment(client, "svc_id", time.Second); err == nil {
		t.Fatal("Error was nil!")
	}
}
```

I expect this to execute many times; each time a `Deployment` is to be checked, `ReadService` must be called in order to get the most recent `Deployment` status, since `Deployments` are objects within `Service`s.